src/components/ChildrenOf.tsx

```diff
@@ -38,32 +38,39 @@ const createList = (children: SilverstripeDocument[]): ReactElement[] => {
     });
 };
 
-const ChildrenOf: StatelessComponent<ChildrenOfProps> = ({ folderName, exclude, currentNode, asList }) => {
+const ChildrenOf: StatelessComponent<ChildrenOfProps> = ({ folderName, exclude, only, currentNode, asList, includeFolders, reverse }) => {
     if (!currentNode) {
         return null;
     }
-    let children: ReactElement[] = [];
-    if (!folderName && !exclude) {
-        const sourceNodes = currentNode.isIndex ? getChildren(currentNode, false) : getSiblings(currentNode)
-        children = asList ? createList(sourceNodes) : createCards(sourceNodes);
+
+    var nodes: SilverstripeDocument[] = [];
+
+    if (!folderName && !exclude && !only) {
```

Comment:
Probably worth adding some inline documentation here to describe what each case means and does.